framework/src/org/checkerframework/common/value/ValueVisitor.java

```diff
@@ -51,29 +83,56 @@ public Void visitAnnotation(AnnotationTree node, Void p) {
         }
 
         Element element = TreeInfo.symbol((JCTree) node.getAnnotationType());
-        if (!(element.toString().equals(ArrayLen.class.getName())
+
+        if (element.toString().equals(IntRange.class.getName())) {
+            // If there are 2 arguments, issue an error if from.greater.than.to.
+            // If there are less than 2 arguments, we needn't worry about this problem because the
+            // other argument would be defaulted to Long.MIN_VALUE or Long.MAX_VALUE accordingly.
+            if (args.size() == 2) {
+                ExpressionTree expFrom;
+                ExpressionTree expTo;
+                AssignmentTree arg0 = (AssignmentTree) args.get(0);
+                AssignmentTree arg1 = (AssignmentTree) args.get(1);
+                if (arg0.getVariable().toString().equals("from")) {
+                    expFrom = arg0.getExpression();
+                    expTo = arg1.getExpression();
+                } else {
+                    expTo = arg0.getExpression();
+                    expFrom = arg1.getExpression();
+                }
+
+                if (isIntLiteral(expFrom.getKind()) && isIntLiteral(expTo.getKind())) {
```

Comment:
How can this not succeed?
If it doesn't succeed, does that indicate an internal problem with the checker?  If so, the checker should halt so we learn about and can fix the problem, rather than silently masking it.
If it's OK for this not to succeed, please add a comment saying why.